src/commands/protocols/get/sender/batch-get-command.js

```diff
@@ -253,78 +253,94 @@ class BatchGetCommand extends Command {
+                            result.responseData.assertions,
+                            blockchain,
+                            paranetNodesAccessPolicy,
+                            contentType,
+                            finalResult,
+                        );
 
-            results.forEach((result) => {
-                if (result.success) {
-                    succsesfulResult.push(result);
-                } else {
-                    failedResults.push(result);
-                }
-            });
+                        for (const [ual, isKCValid] of Object.entries(validationResult)) {
+                            if (isKCValid) {
+                                finalResult.remote[ual] = result.responseData.assertions[ual];
+                                finalResult.metadata[ual] = result.responseData.metadata[ual];
+                                const idx = ualNotPresentLocally.indexOf(ual);
+                                if (idx !== -1) {
+                                    ualNotPresentLocally.splice(idx, 1);
+                                }
+                            }
+                        }
+
+                        if (ualNotPresentLocally.length === 0 && !commandCompleted) {
+                            commandCompleted = true;
+                            await this.operationService.markOperationAsCompleted(
+                                operationId,
+                                blockchain,
+                                finalResult,
+                                [
+                                    OPERATION_ID_STATUS.GET.GET_LOCAL_END,
+                                    OPERATION_ID_STATUS.GET.GET_END,
+                                    OPERATION_ID_STATUS.COMPLETED,
+                                ],
+                            );
+                        }
+                    })
+                    .catch((err) => {
+                        this.logger.warn(`Node ${node.id} failed: ${err.message}`);
+                    }),
+            );
 
-            for (const result of succsesfulResult) {
-                // eslint-disable-next-line no-await-in-loop
-                const validationResult = await this.validateBatchResponse(
-                    result.responseData.assertions,
-                    blockchain,
-                    paranetNodesAccessPolicy,
-                    contentType,
-                    finalResult,
+            // eslint-disable-next-line no-await-in-loop, no-loop-func
+            await new Promise((resolve) => {
+                let settled = 0;
+                // eslint-disable-next-line no-loop-func
+                messagePromises.forEach((p) =>
+                    p.finally(() => {
+                        settled += 1;
+                        if (commandCompleted || settled === promises.length) {
+                            resolve();
+                        }
+                    }),
                 );
+            });
```

Comment:
I would be in favor of using just async/await along with `Promise.allSettled(...)`
Mixing callbacks based API and async/await API makes code less readable imo

I think this all can be replaced with
```js
await Promise.allSettled(
    batch.map(async (node) => {
        try {
            // Main logic here
        } catch (err) {
            this.logger.warn("Node failed...")
        }
    })
)
```